mypy/checker.py

```diff
@@ -3341,12 +3341,24 @@ def check_incompatible_property_override(self, e: Decorator) -> None:
                     self.fail(message_registry.READ_ONLY_PROPERTY_OVERRIDES_READ_WRITE, e)
 
     def visit_with_stmt(self, s: WithStmt) -> None:
+        exceptions_maybe_suppressed = False
         for expr, target in zip(s.expr, s.target):
             if s.is_async:
-                self.check_async_with_item(expr, target, s.unanalyzed_type is None)
+                exit_ret_type = self.check_async_with_item(expr, target, s.unanalyzed_type is None)
             else:
-                self.check_with_item(expr, target, s.unanalyzed_type is None)
-        self.accept(s.body)
+                exit_ret_type = self.check_with_item(expr, target, s.unanalyzed_type is None)
+            if is_literal_type(exit_ret_type, "builtins.bool", False):
+                continue
+            if is_literal_type(exit_ret_type, "builtins.bool", True):
+                exceptions_maybe_suppressed = True
+            elif (isinstance(exit_ret_type, Instance)
+                    and exit_ret_type.type.fullname() == 'builtins.bool'):
+                exceptions_maybe_suppressed = True
+        if exceptions_maybe_suppressed:
+            with self.binder.frame_context(can_skip=True, try_frame=True):
```

Comment:
Although it is almost clear, please add a short comment here attracting attention to `try_frame=True`.